### DISCUSSION TITLE
WSL2 where is docker volume location?

### DISCUSSION THREAD

**Discussion by dryworld** [microsoft/WSL#4176] [General]
* Your Windows build number:  (Type `ver` at a Windows Command Prompt)

Microsoft Windows [Version 10.0.18362.175]
Docker version 18.09.6, build 481bc77

* What you're doing and what's happening:  (Copy&paste the full set of _specific_ command-line steps necessary to reproduce the behavior, and their output. Include screen shots if that helps demonstrate the problem.)

I installed ubuntu-16.04 at WSL2, and also install docker.
I maked volume by docker command.

## docker volume ls
DRIVER              VOLUME NAME
local               doc_vol

* What's wrong / what should be happening instead:

I looking for the mounted location so I execute command next follow

## docker volume inspect doc_vol
[
    {
        "CreatedAt": "2019-06-18T02:11:35Z",
        "Driver": "local",
        "Labels": {},
        "Mountpoint": "/var/lib/docker/volumes/doc_vol/_data",
        "Name": "doc_vol",
        "Options": {},
        "Scope": "local"
    }
]

and then checked the location, It's strange result.

## ls -l /var/lib/docker/volumes/doc_vol/_data
ls: cannot access '/var/lib/docker/volumes/doc_vol/_data': No such file or directory

I checked the '/var/lib/docker/volumes/' only one file exist.

## ls -l /var/lib/docker/volumes/
total 32
-rw------- 1 root root 32768 Jun 18 11:08 metadata.db

where is mounted doc_vol?

**Comment by craigloewen-msft** [comment]
You need to be using `sudo ls` to access the volumes in that folder.

```
caloewen@LOE-WSL:/var/lib/docker$ sudo docker run -d --name=nginxtest --mount source=doc_vol,destination=/usr/share/nginx/html nginx:latest
ec7304a051d3f1564dfa91b4a81538caea236d4b30e0092da90d1737058b145d
caloewen@LOE-WSL:/var/lib/docker$ sudo ls ./volumes/doc_vol/_data
50x.html  index.html
caloewen@LOE-WSL:/var/lib/docker$  
```

**Comment by dryworld** [comment]
thanks @mscraigloewen I already root user.
this editor # change to big bold font. :(

anyway previous BIG font has # prompt
also I try as you said 'sudo ls ./volumes/doc_vol/_data' result is 
'ls: cannot access '/var/lib/docker/volumes/doc_vol/_data': No such file or directory'

**Comment by therealkenc** [comment]
There were no takers on the question and no repro to form a WSL actionable. Might get more eyeballs over at [Superuser](https://superuser.com/). If you post be sure to include more specifics; I (personally) couldn't follow the ask on this one.

**Comment by david-sackstein** [comment]
@craigloewen-msft Your answer is incorrect. sudo does not solve the problem because it is not caused by lack of permissions. That path does not exist.

**Comment by craigloewen-msft** [comment]
@david-sackstein you're totally correct!

From my answer, it seems that I wasn't able to repro the error, which means it could be something specific to your setup. I agree with @therealkenc , as of right now it's hard for us to try and repro this so superuser might be your best choice, or posting more repro steps here for us to try. 

Thanks!

**Comment by kerams** [comment]
With Docker Desktop 2.1.7.0 and WSL 2 (Ubuntu), I can see the volumes at /mnt/wsl/docker-desktop-data/data/docker/volumes. Access requires elevated permissions, so you could do `ubuntu.exe config --default-user root` before entering wsl.

**Reply by Serhioromano** [reply]
I do not have this folder as well as var/lib/docker

**Reply by vikasedu10** [reply]
@Serhioromano, are you using a mac or Windows WSL/WSL2?

**Comment by mahmoud-samy** [comment]
@kerams Thanks.

From linux:  `sudo ls /mnt/wsl/docker-desktop-data/data/docker/volumes`
From windows: `\\wsl$\docker-desktop-data\mnt\wsl\docker-desktop-data\data\docker\volumes`

**Reply by Serhioromano** [reply]
I have found it in `\\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes`

**Reply by shashi2k6** [reply]
I agree

**Reply by PhoenixNest** [reply]
Hello guys, i have met this problem as well, but there is nothing in my folder

```bash
ubuntu@asdasd:/mnt$ ls
c  d  g  w  wsl  x  y  z
ubuntu@asdasd:/mnt$ cd wsl/
ubuntu@asdasd:/mnt/wsl$ ls
docker-desktop  docker-desktop-bind-mounts  docker-desktop-data
ubuntu@asdasd:/mnt/wsl$ cd docker-desktop-data/
ubuntu@asdasd:/mnt/wsl/docker-desktop-data$ ls
data  isocache  tarcache  version-pack-data
ubuntu@asdasd:/mnt/wsl/docker-desktop-data$ cd data/
ubuntu@asdasd:/mnt/wsl/docker-desktop-data/data$ ls
ubuntu@asdsad:/mnt/wsl/docker-desktop-data/data$ ls -a
.  ..
ubuntu@asdasd:/mnt/wsl/docker-desktop-data/data$ sudo ls -a
.  ..
ubuntu@asdasd:/mnt/wsl/docker-desktop-data/data$ cd ../
ubuntu@asdsad:/mnt/wsl/docker-desktop-data$ ls
data  isocache  tarcache  version-pack-data
ubuntu@asdasd:/mnt/wsl/docker-desktop-data$ cd version-pack-data/
ubuntu@asdsad:/mnt/wsl/docker-desktop-data/version-pack-data$ ls
ubuntu@asdasd:/mnt/wsl/docker-desktop-data/version-pack-data$
```

**Reply by IARI** [reply]
same problem as @PhoenixNest .

**Reply by Venryx** [reply]
For people reading this discussion trying to find where their project's files are being output, from docker image-builds: In my case at least, my project files were not being output to the `\\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes` folder mentioned above, but rather the `\\wsl$\docker-desktop-data\version-pack-data\community\docker\overlay2\XXX\diff` folders.

I found this out by using [grepWin](https://tools.stefankueng.com/grepWin.html) to search within the `\\wsl$\docker-desktop-data\version-pack-data\community\` folder for my project files that I know were included. (I know because I used the docker [dive](https://github.com/wagoodman/dive) tool to inspect its contents -- but wanted to know the "actual path" since the dive tool takes a while to load)

Screenshot:
![](https://i.imgur.com/r1c06Ex.png)

**Reply by masterflitzer** [reply]
you are such a life saver

**Comment by haimat** [comment]
@mahmoud-samy Thanks for this info.
But how can I find out where Docker Desktop saves this data on the Windows host?

**Comment by kerams** [comment]
@haimat, it might be in in a VHDX file in `C:\Users\_user_\AppData\Local\Docker\wsl\data`

**Comment by haimat** [comment]
@kerams Ohh yes, that's it, thanks a lot!
Now, do you also know whether I can `--export` & `--import` both WSL's to another location, or does that confuse Docker Desktop? I know this is not the Docker GitHub repo, but maybe you do know that :-)

**Reply by ghost** [reply]
maybe will not, because docker communicate with wsl, 
then export/import is controled by wsl, wsl know data, wsl provide service for docker, docker need to know things under wsl

**Comment by thejacket** [comment]
Thanks @mahmoud-samy

So what those paths really mean if they're not referring to host machine? (excerpt from Docker Dashboard, same as CLI's docker volume inspect)
![image](https://user-images.githubusercontent.com/23661765/78978919-0d8e8980-7b1b-11ea-82d5-17e51b88e724.png)

**Comment by alexandermalfait** [comment]
The key seems to be the "virtual" wsl distro named `docker-desktop-data`

See https://dev.to/kimcuonthenet/move-docker-desktop-data-distro-out-of-system-drive-4cg2 for solution

**Comment by mahmoud-samy** [comment]
@haimat  @thejacket Sorry for the late reply

AFAIK, when you use WSL integration, two distros will be created for you (docker-desktop & docker-desktop-data)
![image](https://user-images.githubusercontent.com/3718630/88474617-b6c18a00-cf28-11ea-8f27-f685d22788f8.png)

To reach their file system, you need to access it over network
```
\\wsl$\docker-desktop
\\wsl$\docker-desktop-data
```

**Comment by dmckee-ebsco** [comment]
This is... not simple. I don't know why it's this hard. I found what I was looking for after dropping into wsl session, and using the `find` command to locate a folder name that I knew...

```bash
find /mnt/host/wsl/docker-desktop-data/data/ -name 'jekyll'
```

**Comment by filipp-codes** [comment]
I found my volume here:
`\\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes`

**Reply by ZealousMacwan** [reply]
For WSL2 and manual installation of Ubuntu-20.04 distro, This worked for me !

**Reply by PhoenixNest** [reply]
This is useful for my windows wsl2 ubuntu, thanks bro.

**Reply by mozarik** [reply]
How to open that and access that file from my default Distros (eg:Ubuntu 20.04) terminal. I tried to run that wsl image `wsl -d docker-desktop-data` but nothing happen.

**Reply by PhoenixNest** [reply]
> How to open that and access that file from my default Distros (eg:Ubuntu 20.04) terminal. I tried to run that wsl image `wsl -d docker-desktop-data` but nothing happen.

you can use the “add internet location” in your default windows files manager, and that access to docker

**Reply by mozarik** [reply]
> you can use the “add internet location” in your default windows files manager, and that access to docker

Thanks. After that i have to add the drive using 
```
$ sudo mkdir /mnt/e
$ sudo mount -t drvfs E: /mnt/e
```

**Reply by usister** [reply]
It worked! Thank you!

**Reply by GabrielFMPinheiro** [reply]
It worked! Thank you!

**Comment by ethzero** [comment]
> I found my volume here:
> `\\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes`

This took far too much failed Docker documentation trawling and Googling to find this answer. Nice one!

**Reply by IARI** [reply]
Nothing there for me either.

**Comment by sandy-rt** [comment]
> I found my volume here:
> `\\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes`

Thanks, buddy, it helps a lot.

**Comment by gaia** [comment]
very helpful thread!

though i want to know where the docker containers reside from the windows perspective.

it seems to be inside the vhdx at `C:\Users\_johndoe_\AppData\Local\Packages\_distro_\LocalState` 

is this correct? if I backup this VHDX from windows I will have a backup of the docker containers (in addition to the _distro_ I use)?

This is why it worries me whether the backup will work. I can't see the files from within linux

![image](https://user-images.githubusercontent.com/87547/96352459-1cc93380-1078-11eb-9dbc-08e6eb43391c.png)

but they are there via windows

![image](https://user-images.githubusercontent.com/87547/96352482-471af100-1078-11eb-950d-4fa239c63b28.png)

**Comment by cdalexndr** [comment]
This should be clearly documented, as the advantage of WSL is that you can access volume files directly

**Comment by sebtiz13** [comment]
> I found my volume here:
> `\\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes`

Thanks, This took far search to find this.
Just i want truncate log of containers but like said @gaia the mounted directory in WSL are not accessible
`/wsl/docker-desktop-data/version-pack-data` are empty

Somebody have an idea, how can access to this file from WSL2 ?

**Reply by ghost** [reply]
Exact same use case here, truncating container logs.
I can see `\\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes` populated with container information (including logs) when using Windows Explorer.

No dice when accessing from WSL, same result: `/wsl/docker-desktop-data/version-pack-data` is empty when viewed from within WSL.

**Comment by spicylentils** [comment]
Counter intuitive why the path would include 'version-pack-data\community' !!! Why should this path have anything to do with the names associated with these directories??? Docker/Microsoft almost lost a user of Docker Desktop for Windows. I spent way too much time trying to find out where my volumes were stored. Not developer friendly!!!

**Comment by pauloricardomg** [comment]
`/mnt/wsl/docker-desktop-data/version-pack-data` is empty and `docker-desktop-data\version-pack-data\community\docker\volumes` does not exist, so I'm still not able to find where my images are located. WSL2 and docker desktop 2.4.2.0. Does anyone know where to find it?

**Comment by spicylentils** [comment]
Did you try prefixing your path with: 

\\\\wsl$\

?

**Comment by Kallestofeles** [comment]
> `/mnt/wsl/docker-desktop-data/version-pack-data` is empty and `docker-desktop-data\version-pack-data\community\docker\volumes` does not exist, so I'm still not able to find where my images are located. WSL2 and docker desktop 2.4.2.0. Does anyone know where to find it?

This is strange. I get the exact same behavior and am unable to see the directory when in WSL2 guest. But when accessing the same folder in windows explorer using `\\wsl$\` I see quite a few folders and the volumes are listed there nicely as well. Why is this a thing and we cannot see the folder in WSL guest?

**Comment by truboxl** [comment]
`\\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes` is now empty with a `metadata.db` file

`\\wsl$\docker-desktop-data\version-pack-data\community\docker\overlay2` \ randomstring \ `diff` folder
is where you should look for

Not sure how that randomstring is calculated
It's not container ID or image ID
Just brute force your way using search or `find` or other tools
You also can use `docker image inspect [container_name]` based on this [answer](https://stackoverflow.com/questions/62380124/where-docker-image-is-stored-with-docker-desktop-for-windows/63441376#63441376)

I am on `Docker version 19.03.13, build 4484c46d9d` or `Docker Desktop 2.5.0.1`

**Reply by Venryx** [reply]
Yeah, this is where I found my project's files output as well. (it seems the OS files are in the `volumes` folder, but your project-specific files go in the `overlay2` folders)

**Comment by ekam230** [comment]
\\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes

**Comment by johnnyoshika** [comment]
From powershell, I can get to my volumes like this:

```
cd \\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes
```

I can also type this in the Windows Explorer address to get to the same path:
```
\\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes
```

I can't seem to access that folder from WSL2 though. Anyone know the path? Is it possible?

The reason why I want to access that path from WSL is because I'd like to use bash commands to play with the volumes (e.g. back up, restore, etc).

**Comment by evertonstz** [comment]
This is insanity lol.
Anyone found a way I can reach the volumes from the Linux side? I can actually find it via PowerShell at `\\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes`.

**Comment by spicylentils** [comment]
1. In windows:
    net use w: \\\\wsl$\\Ubuntu
2. In WSL2
    sudo mkdir /mnt/w
    sudo mount -t drvfs w: /mnt/w
    ls /mnt/w

**Reply by evertonstz** [reply]
`mount: /mnt/w: special device w: does not exist.`

**Reply by run0ut** [reply]
> `mount: /mnt/w: special device w: does not exist.`

I assume, that you might have missed the first part: connect net drive in Windows. At least that was the case for me.

**Reply by run0ut** [reply]
That worked, thank you! My case was a little diffrent, i needed access to containers' logs to experiment with ELK stack and filebeat, though the solution is the same.

**Reply by Libianno** [reply]
Thanks, it worked for me!

**Reply by PramSin** [reply]
That really works, thx so much!

**Comment by thejacket** [comment]
I just moved to bare metal Ubuntu, solved.

**Comment by Serhioromano** [comment]
I've read this thread but found no answer. In my porteiner UI I have this for volume.

![2021-01-06_20-49-38](https://user-images.githubusercontent.com/650741/103781666-d7a61380-5060-11eb-8908-d3d22b34d01a.png)

But when I login to WSL2 it does not has this volume. This path does not exists. I've looked in suggested pass.

My main concern is DB. I have mysql docker image and if I have to reinstall OS I want this volume to be saved with windows file history feature. Or NAS backup. That is why I am looking where actual data is.

**Reply by ZealousMacwan** [reply]
Did you try this command ? 
`\\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes`

**Comment by cveld** [comment]
Can anyone from the product team reply to our questions?
- How can we query the wsl2 / docker server / host to get the real location? 
- How is this stub location `/var/lib/docker` being faked? 
- Why is the contents only visible from the Windows side?

**Comment by swr71999** [comment]
Maybe in WSL2 way, data only exists in Windows systems.
When we use WSL2 based engine, Docker Desktop mount "docker.sock" into Linux "/var/run/docker.sock". Linux docker client access to windows docker service.
I find this idea in below link: https://blog.csdn.net/qqhappy8/article/details/106819429

**Comment by djl2** [comment]
It's been almost 2 years with no clear answer on this issue. Can someone from Microsoft please respond?

**Reply by ZealousMacwan** [reply]
Try this on your explorer, It works for me.
`\\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes`

**Reply by a26796879** [reply]
It works for me too

**Reply by sunilgaddi** [reply]
yes it works but , but in inspect volume it shows /var/lib/docker/.....

but i dont have docker folder in the lib directory

can anyone help please

**Comment by stephen-turner** [comment]
Hello everyone, this is a Docker response.

There are several correct answers above about where the files are stored, and you can read them there. However, that is not an official interface, so we don't guarantee it in future. The recommended way to backup volumes is by mounting them into another container as described in our documentation: https://docs.docker.com/storage/volumes/#backup-restore-or-migrate-data-volumes.

We are also considering building some volume manipulation functionality into the Docker Desktop GUI, but we don't have it yet.

**Comment by nvRelax** [comment]
> Prerequisites: wsl2 installed, ubuntu installed

A buddy above talked about how to mount docker's volumes directory into ubuntu so that you can read the contents of these volumes directly in ubuntu. But it's not very clear, so I'll recap what I did.

1. Step 1.
> In windows command line, execute the command: `net use h: \\wsl$\docker-desktop-data`

Where, please replace `h:` with a drive letter that does not exist on your computer. After this step, you can see a `network location` in windows explorer

2. Step 2.
Go to ubuntu and execute the command.
> sudo mkdir /mnt/docker
> sudo mount -t drvfs h: /mnt/docker

Then you can manipulate docker's volume in the `/mnt/my-docker/version-pack-data/community/docker/volumes` directory.

**Reply by ChristianOConnor** [reply]
> 
> 
> > Prerequisites: wsl2 installed, ubuntu installed
> 
> A buddy above talked about how to mount docker's volumes directory into ubuntu so that you can read the contents of these volumes directly in ubuntu. But it's not very clear, so I'll recap what I did.
> 
>     1. Step 1.
> 
> 
> > In windows command line, execute the command: `net use h: \\wsl$\docker-desktop-data`
> 
> Where, please replace `h:` with a drive letter that does not exist on your computer. After this step, you can see a `network location` in windows explorer
> 
>     1. Step 2.
>        Go to ubuntu and execute the command.
> 
> 
> > sudo mkdir /mnt/docker
> > sudo mount -t drvfs h: /mnt/docker
> 
> Then you can manipulate docker's volume in the `/mnt/my-docker/version-pack-data/community/docker/volumes` directory.

I did your steps exactly and I got this  
```
mount: /mnt/docker: special device h: does not exist.
<3>init: (104) ERROR: UtilCreateProcessAndWait:489: /bin/mount failed with status 0x2000
<3>init: (104) ERROR: MountPlan9:478: mount cache=mmap,rw,trans=fd,rfdno=3,wfdno=3,msize=65536,aname=drvfs;path=h:;symlinkroot=/mnt/ failed 2
No such file or directory
```  
  
TThis error was when I did the command `sudo mount -t drvfs h: /mnt/docker`

**Reply by yhslai** [reply]
> > Prerequisites: wsl2 installed, ubuntu installed
> 
> A buddy above talked about how to mount docker's volumes directory into ubuntu so that you can read the contents of these volumes directly in ubuntu. But it's not very clear, so I'll recap what I did.
> 
> 1. Step 1.
> 
> > In windows command line, execute the command: `net use h: \\wsl$\docker-desktop-data`
> 
> Where, please replace `h:` with a drive letter that does not exist on your computer. After this step, you can see a `network location` in windows explorer
> 
> 1. Step 2.
>    Go to ubuntu and execute the command.
> 
> > sudo mkdir /mnt/docker
> > sudo mount -t drvfs h: /mnt/docker
> 
> Then you can manipulate docker's volume in the `/mnt/my-docker/version-pack-data/community/docker/volumes` directory.

Omg finally. After literally years, someone found the solution. I don't know what's going on with WSL and Docker's team.

**Reply by ducn** [reply]
This issue supported now, using this command in wsl:

`sudo mount -t drvfs '\\wsl$\docker-desktop-data' /mnt/wsl2`

then you can access docker-desktop-data in wsl from /mnt/wsl2

Please refer this document: https://docs.microsoft.com/en-us/archive/blogs/wsl/file-system-improvements-to-the-windows-subsystem-for-linux.

**Reply by ChristianOConnor** [reply]
Oh awesome thanks! Glad it's supported now!

**Reply by willburden** [reply]
I have the same problem as @ChristianOConnor. How did you solve it?

**Reply by lyf9528** [reply]
thanks so much!!

**Comment by thusinh1969** [comment]
Follow this git https://github.com/DamionGans/ubuntu-wsl2-systemd-script
Once done, you should be able to install microk8s OK and then full cluster can be created and run well. That is the way I got around.

Steve

**Comment by ducn** [comment]
Please check answer at https://github.com/microsoft/WSL/discussions/4176#discussioncomment-1458301

**Comment by moisesjuniorops** [comment]
Hi guys! I have the same problem here and i solved it trying this:

I use Ubuntu 20.04 as a subsystem on WSL 2;
I closed the Ubuntu and opened my Windows PowerShell as administrator and run this command:
`wsl --shutdown` and it has closed all of my applications running.
I opened my Ubuntu for the second time and as i'm trying to run docker commands such as root or with my user i kept getting errors from daemon;

Then i ran `sudo dockerd` to start the docker daemon.

It has started docker again and i created a new volume to located on his path, now located on the referenced on the "docker volume inpect" located in: `var/lib/docker/volumes`.

I hope i can help some of you guys!

**Comment by gifflet** [comment]
I have found my volumes in `\\wsl$\docker-desktop-data\data\docker\volumes`

**Reply by shrikantpardhi** [reply]
Docker version 20.10.+ uses `\\wsl$\docker-desktop-data\data\docker\volumes`. The previous version used a different directory.

**Comment by pavlokolodka** [comment]
Hi everyone! 

I am using WLS 2 (Ubuntu-18.04, Docker version 20.10.20) and I also encountered this problem.

I found that there are two ways to locate the volume file:
In my case, I needed to save a PostgreSQL volume, so I took it as an example.

If you save your volume as follows `-v postgres-data:/var/lib/postgresql/data `
you can find it as mentioned before `\\wsl$\docker-desktop-data\data\docker\volumes`
Also, if you are using Docker Desktop, you can see that the Volumes tab has new volume

But if you save your volume with "/" before your volume path, e.g. `-v /postgres-data:/var/lib/postgresql/data` 
then your file will be saved in `\\wsl$\docker-desktop-data\mnt\wsl\docker-desktop-bind-mounts\Ubuntu-18.04 ` (folder  name is hashed)
also it duplicated in  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; `\\wsl$\Ubuntu-18.04\mnt\wsl\docker-desktop-bind-mounts\Ubuntu-18.04` (folder name is hashed)
and &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; `\\wsl$\Ubuntu-18.04` (real folder name)
And in Docker Desktop nothing was created in the Volumes tab

I hope this was helpful.

**Reply by AnakinShieh** [reply]
I meet the same problem, but I didn't find the corresponding location as yours. It's kind of strange and I guess it's revelant to the windows desktop.

**Reply by asimplepeople** [reply]
I meet the same problem。

**Comment by pedrohfborges** [comment]
\\wsl$\docker-desktop-data\data\docker\volumes

**Comment by sick-pupil** [comment]
I find the volumes mount path is '\wsl.localhost\docker-desktop-data\data\docker\volumes', with Docker Desktop 4.18.0

**Comment by xhiroga** [comment]
To make Docker volumes accessible in WSL2 just like in a default Ubuntu installation, these steps work for me:

1. In your WSL2 Ubuntu terminal, run:

```shell
sudo mkdir -p /var/lib/docker
sudo mount -t drvfs '\\wsl.localhost\docker-desktop-data\data\docker' /var/lib/docker
```
Note: This path is verified with Docker Desktop 4.34.3 (Engine 27.2.0). Older versions might use a different path.

2. Now you can access Docker volumes at `/var/lib/docker/volumes`, just as you would in a standard Ubuntu installation.
3. To verify the mount was successful, run:

```
ls /var/lib/docker/volumes
docker volume ls
```

You should see your Docker volumes listed in both outputs.

**Comment by abritabroad** [comment]
as of Docker v4.36 the new location of the volumes can be be mounted thus:

`sudo mount -t drvfs '\\wsl.localhost\docker-desktop\mnt\docker-desktop-disk\data\docker\volumes' /var/lib/docker`

**Reply by xhiroga** [reply]
Thanks!
I guess `sudo mount -t drvfs '\\wsl.localhost\docker-desktop\mnt\docker-desktop-disk\data\docker' /var/lib/docker` is better. It make seeing `/var/lib/docker/volumes` possible and it's equivalent to Linux.

**Comment by pwnht** [comment]
just run ` docker run -v /:/rootfs -it ubuntu:20.04 bash` and edit it like linux like this
`
root@27e9cef2b051:/# ls /rootfs/var/lib/docker/containers/

1e3290c1c31db60b4b094d97d6e6a0b695ff9c799cbd533e0990a06affc5c4d2
27e9cef2b0518db71cef880f55704170e596e41661cde7502028d611a838c04b
2f7b378366b2fe684e17a9f3f8657f1031d8fc05d5c0f2c21ee5d11dcda7defa
fd04921a65985c8a524b808f72648572f70d22443fbb0755ae485dfabe138416
`

**Comment by wengtiancheng** [comment]
I can't get access to `docker-desktop-data` on  my computer. But I found the volumes in `\\wsl.localhost\docker-desktop\mnt\docker-desktop-disk\data\docker\volumes`
